synapse/storage/util/id_generators.py

```diff
@@ -473,7 +474,7 @@ def _mark_id_as_finished(self, next_id: int):
 
                 finished = set()
 
-                min_unfinshed = min(self._unfinished_ids)
+                min_unfinshed = self._unfinished_ids[0]
```

Comment:
If we only care about the minimum one, would it make more sense to track that separately? (Does `SortedSet` have any overhead compared to `set`?)

Reply:
Potentially, but in practice this change stopped the ID generators from appearing in the profiling, so :shrug: